### ISSUE TITLE
Expose CLI functionality as a separate class

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [palantir/tslint#1675]
_**Context**: I'm working on [autotslint](https://github.com/autolint/autotslint), a CLI tool to take in TSLint failures and, when possible, fix them in the files._

_**Ask**_: Can the existing logic in `tslint-cli.ts` be exposed publicly as a usable class and documented in the readme? Ideally we should be able to do something like the following:

Autotslint will need to be able to generate "waves" of TSLint failures: given the standard configurations that can be given via tslint-cli, it should generate failures for rules. It's possible to do this using the existing require("tslint") linter and some work, but would be much simpler if the existing CLI logic could be re-used.

```typescript
import {LintCli} from "tslint/lib/lintCli");

const options = {
    exclude: "**/*.d.ts",
    rules: {
        "variable-name": true,
        "quotemark": [true, "double"]
    }
};

const linter = new LintCli(options);

const result = linter.processFiles(["**/*.ts"]/*, ts.createProgram(...) */);
```

The CLI class would just be a refactor of tslint-cli.ts, not a change in any existing behavior. That being said, it would be nice to switch the options to have shorthand as aliases so the interface could use strings like `"config"` instead of `"c"`.

**Comment by adidahiya** [comment]
Hey, I don't know if you noticed this (there are no docs...) but we have started implementing auto-fixing in tslint itself. This is on the master branch and in the 4.0.0-dev.* releases. See https://github.com/palantir/tslint/pull/1423, https://github.com/palantir/tslint/pull/1574

**Comment by JoshuaKGoldberg** [comment]
D:

**Comment by JoshuaKGoldberg** [comment]
Still - can this issue remain open? It could be useful to call the CLI programmatically.
